crates/bevy_window/src/lib.rs

```diff
@@ -118,6 +118,16 @@ impl Plugin for WindowPlugin {
         if settings.close_when_requested {
             app.add_system(close_when_requested);
         }
+
+        #[cfg(debug_assertions)]
+        {
+            app.add_system(|wd: Option<Res<WindowDescriptor>>, mut once: Local<bool>| {
+                if !*once && wd.is_some() {
+                    warn!("The WindowDescriptor resource must be inserted before the WindowPlugin is added. Make sure to insert WindowDescriptor before adding the DefaultPlugins.");
```

Comment:
nit: I'd suggest splitting the message in several lines using 
```suggestion
                    warn!(
                        "The WindowDescriptor resource must be inserted before the WindowPlugin is added. \
                        Make sure to insert WindowDescriptor before adding the DefaultPlugins.",
                    );
```

Reply:
I thought using a \ would only remove the newline but it nicely omits all the whitespace. Neat.